### ISSUE TITLE
not building on latest plasma

### ISSUE THREAD

**Issue by illnesse** [taj-ny/kwin-effects-forceblur#106]
```
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Installing in the same prefix as Qt, adopting their path scheme.
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Found OpenGL: /usr/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /usr/lib/libxkbcommon.so (found suitable version "1.7.0", minimum required is "0.5.0")
-- Found WrapVulkanHeaders: /usr/include
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found KF6Config: /usr/lib/cmake/KF6Config/KF6ConfigConfig.cmake (found version "6.6.0")
-- Found KF6ConfigWidgets: /usr/lib/cmake/KF6ConfigWidgets/KF6ConfigWidgetsConfig.cmake (found version "6.6.0")
-- Found KF6CoreAddons: /usr/lib/cmake/KF6CoreAddons/KF6CoreAddonsConfig.cmake (found version "6.6.0")
-- Found KF6Crash: /usr/lib/cmake/KF6Crash/KF6CrashConfig.cmake (found version "6.6.0")
-- Found KF6GlobalAccel: /usr/lib/cmake/KF6GlobalAccel/KF6GlobalAccelConfig.cmake (found version "6.6.0")
-- Found Gettext: /usr/bin/msgmerge (found version "0.22.5")
-- Found KF6I18n: /usr/lib/cmake/KF6I18n/KF6I18nConfig.cmake (found version "6.3.0")
-- Found X11: /usr/include
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found KF6KIO: /usr/lib/cmake/KF6KIO/KF6KIOConfig.cmake (found version "6.6.0")
-- Found KF6Service: /usr/lib/cmake/KF6Service/KF6ServiceConfig.cmake (found version "6.6.0")
-- Found KF6Notifications: /usr/lib/cmake/KF6Notifications/KF6NotificationsConfig.cmake (found version "6.6.0")
-- Found KF6WidgetsAddons: /usr/lib/cmake/KF6WidgetsAddons/KF6WidgetsAddonsConfig.cmake (found version "6.6.0")
-- Found KF6WindowSystem: /usr/lib/cmake/KF6WindowSystem/KF6WindowSystemConfig.cmake (found version "6.6.0")
-- Found KF6GuiAddons: /usr/lib/cmake/KF6GuiAddons/KF6GuiAddonsConfig.cmake (found version "6.6.0")
-- Found KF6KCMUtils: /usr/lib/cmake/KF6KCMUtils/KF6KCMUtilsConfig.cmake (found version "6.6.0")
-- Found KF6: success (found suitable version "6.3.0", minimum required is "5.240.0") found components: Config ConfigWidgets CoreAddons Crash GlobalAccel I18n KIO Service Notifications Service WidgetsAddons WindowSystem GuiAddons KCMUtils
-- Found epoxy: /usr/lib/libepoxy.so (found version "1.5.10")
-- Found XCB_XCB: /usr/lib/libxcb.so (found version "1.17.0")
-- Found XCB: /usr/lib/libxcb.so (found version "1.17.0") found components: XCB
-- Found Wayland_Server: /usr/lib/libwayland-server.so (found version "1.23.1")
-- Found Wayland: /usr/lib/libwayland-server.so (found version "1.23.1") found components: Server
-- The following OPTIONAL packages have been found:

* OpenGL
* Qt6UiTools (required version >= 6.6.0)
* Qt6WidgetsTools (required version >= 6.8.0)
* Qt6Xml (required version >= 6.5.0)
* Qt6Network (required version >= 6.5.0)
* Qt6DBusTools (required version >= 6.8.0)
* Qt6QmlTools (required version >= 6.8.0)
* Qt6CoreTools (required version >= 6.8.0)
* XKB (required version >= 0.5.0), XKB API common to servers and clients., <http://xkbcommon.org>
* WrapVulkanHeaders
* Qt6GuiTools (required version >= 6.8.0)

-- The following REQUIRED packages have been found:

* ECM
* Qt6 (required version >= 6.6.0)
* KF6Crash (required version >= 5.240.0)
* KF6GlobalAccel (required version >= 5.240.0)
* Gettext
* KF6I18n (required version >= 5.240.0)
* KF6KIO (required version >= 5.240.0)
* KF6Notifications (required version >= 5.240.0)
* KF6Service (required version >= 5.240.0)
* KF6WidgetsAddons (required version >= 5.240.0)
* KF6GuiAddons (required version >= 5.240.0)
* KF6ConfigWidgets (required version >= 6.6.0)
* KF6KCMUtils (required version >= 5.240.0)
* KF6 (required version >= 5.240.0)
* XCB, X protocol C-language Binding, <https://xcb.freedesktop.org/>
* Qt6Gui (required version >= 6.5.0)
* X11
* epoxy
* KWin
* KDecoration2
* KWinDBusInterface

-- Configuring done (1.4s)
-- Generating done (0.0s)
-- Build files have been written to: /home/user/.cache/yay/kwin-effects-forceblur-git/src/build
make: Entering directory '/home/user/.cache/yay/kwin-effects-forceblur-git/src/build'
[  0%] Built target forceblur_autogen_timestamp_deps
[  0%] Built target kwin_better_blur_config_autogen_timestamp_deps
[  4%] Automatic MOC for target kwin_better_blur_config
[  9%] Automatic MOC for target forceblur
AutoMoc warning
---------------
"SRC:/src/main.cpp"
includes the moc file "main.moc", but does not contain a Q_OBJECT, Q_GADGET, Q_NAMESPACE, Q_NAMESPACE_EXPORT, Q_GADGET_EXPORT, Q_ENUM_NS, K_PLUGIN_FACTORY, K_PLUGIN_CLASS, K_PLUGIN_FACTORY_WITH_JSON or K_PLUGIN_CLASS_WITH_JSON macro.

[  9%] Built target kwin_better_blur_config_autogen
[ 14%] Generating kwineffects_interface.cpp, kwineffects_interface.h
[ 23%] Generating ui_blur_config.h
[ 23%] Generating blurconfig.h, blurconfig.cpp
[ 28%] Generating moc_kwineffects_interface.cpp
[ 28%] Built target forceblur_autogen
[ 38%] Automatic RCC for blur.qrc
[ 38%] Generating blurconfig.h, blurconfig.cpp
[ 42%] Building CXX object src/CMakeFiles/forceblur.dir/blurconfig.cpp.o
[ 47%] Building CXX object src/CMakeFiles/forceblur.dir/forceblur_autogen/mocs_compilation.cpp.o
[ 57%] Building CXX object src/CMakeFiles/forceblur.dir/blur.cpp.o
[ 57%] Building CXX object src/CMakeFiles/forceblur.dir/main.cpp.o
[ 61%] Building CXX object src/CMakeFiles/forceblur.dir/forceblur_autogen/EWIEGA46WW/qrc_blur.cpp.o
[ 66%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/blurconfig.cpp.o
[ 71%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/blur_config.cpp.o
[ 76%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/kwineffects_interface.cpp.o
[ 80%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/kwin_better_blur_config_autogen/mocs_compilation.cpp.o
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp: In member function ‘virtual void KWin::BlurEffect::reconfigure(KWin::Effect::ReconfigureFlags)’:
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp:240:47: warning: unused parameter ‘flags’ [-Wunused-parameter]
240 | void BlurEffect::reconfigure(ReconfigureFlags flags)
|                              ~~~~~~~~~~~~~~~~~^~~~~
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp: In member function ‘QRegion KWin::BlurEffect::decorationBlurRegion(const KWin::EffectWindow*) const’:
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp:570:70: error: ‘const class KWin::EffectWindow’ has no member named ‘decorationInnerRect’
570 |     QRegion decorationRegion = QRegion(w->decoration()->rect()) - w->decorationInnerRect().toRect();
|                                                                      ^~~~~~~~~~~~~~~~~~~
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp: In member function ‘QRegion KWin::BlurEffect::blurRegion(KWin::EffectWindow*) const’:
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp:591:91: error: ‘class KWin::EffectWindow’ has no member named ‘decorationInnerRect’
591 |                 region += content->translated(w->contentsRect().topLeft().toPoint()) & w->decorationInnerRect().toRect();
|                                                                                           ^~~~~~~~~~~~~~~~~~~
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp: In member function ‘void KWin::BlurEffect::blur(const KWin::RenderTarget&, const KWin::RenderViewport&,KWin::EffectWindow*, int, const QRegion&, KWin::WindowPaintData&)’:
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp:992:28: warning: loop variable ‘rect’ of type ‘const QRectF&’ binds to a temporary constructed from type‘const QRect’ [-Wrange-loop-construct]
992 |         for (const QRectF &rect : effectiveShape) {
|                            ^~~~
/home/user/.cache/yay/kwin-effects-forceblur-git/src/kwin-effects-forceblur-git/src/blur.cpp:992:28: note: use non-reference type ‘const QRectF’ to make the copy explicit or ‘const QRect&’ to prevent copying
make[2]: *** [src/CMakeFiles/forceblur.dir/build.make:124: src/CMakeFiles/forceblur.dir/blur.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:714: src/CMakeFiles/forceblur.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

```
qt5-base 5.15.15+kde+r127-1
qt5-declarative 5.15.15+kde+r25-1
qt5-graphicaleffects 5.15.15-1
qt5-multimedia 5.15.15+kde+r2-1
qt5-quick3d 5.15.15+kde+r1-1
qt5-quickcontrols 5.15.15-1
qt5-quickcontrols2 5.15.15+kde+r5-1
qt5-script 5.15.17-3
qt5-sensors 5.15.15-1
qt5-speech 5.15.15+kde+r1-1
qt5-svg 5.15.15+kde+r5-1
qt5-tools 5.15.15+kde+r3-2
qt5-translations 5.15.15-1
qt5-wayland 5.15.15+kde+r59-1
qt5-x11extras 5.15.15-1
qt6-5compat 6.8.0beta4-1
qt6-base 6.8.0beta4-1
qt6-charts 6.8.0beta4-1
qt6-connectivity 6.8.0beta4-1
qt6-declarative 6.8.0beta4-1
qt6-imageformats 6.8.0beta4-1
qt6-multimedia 6.8.0beta4-1
qt6-multimedia-ffmpeg 6.8.0beta4-1
qt6-positioning 6.8.0beta4-1
qt6-quick3d 6.8.0beta4-1
qt6-quicktimeline 6.8.0beta4-1
qt6-sensors 6.8.0beta4-1
qt6-shadertools 6.8.0beta4-1
qt6-speech 6.8.0beta4-1
qt6-svg 6.8.0beta4-1
qt6-tools 6.8.0beta4-1
qt6-translations 6.8.0beta4-1
qt6-virtualkeyboard 6.8.0beta4-1
qt6-wayland 6.8.0beta4-1
qt6-webchannel 6.8.0beta4-1
qt6-webengine 6.8.0beta4-1
qt6-websockets 6.8.0beta4-1
qt6-webview 6.8.0beta4-1
```

**Comment by theolaos** [comment]
Same here (fedora 40)

**Comment by taj-ny** [comment]
Fixed in v1.3.2.
